Realm/RLMSyncSession.h

```diff
@@ -30,11 +32,59 @@ typedef NS_ENUM(NSUInteger, RLMSyncSessionState) {
     RLMSyncSessionStateInvalid
 };
 
+/**
+ The type of data transfer that a particular progress notification block will track.
+
+ Progress notification blocks can be registered on sessions if your app wishes to be informed
+ how many bytes have been uploaded or downloaded, for example to show progress indicator UIs.
+ */
+typedef NS_ENUM(NSUInteger, RLMSyncProgressDirection) {
+    /// For monitoring upload progress.
+    RLMSyncProgressDirectionUpload,
+    /// For monitoring download progress.
+    RLMSyncProgressDirectionDownload,
+};
+
+/**
+ The desired behavior of a progress notification block.
+
+ Progress notification blocks can be registered on sessions if your app wishes to be informed
+ how many bytes have been uploaded or downloaded, for example to show progress indicator UIs.
+ */
+typedef NS_ENUM(NSUInteger, RLMSyncProgress) {
+    /// The block will be called forever, or until the user unregisters it.
+    /// It will always report the latest number of transferred bytes, and the
+    /// latest number of total bytes to be transferred.
+    RLMSyncProgressReportIndefinitely,
+    /// The block will, upon registration, store the total number of bytes
+    /// to be transferred. When invoked, it will always report the latest number
+    /// of transferred bytes out of that original number of bytes to be
+    /// transferred. When the number of transferred bytes reaches or exceeds the
+    /// number of bytes to be transferred, the block will be unregistered.
+    RLMSyncProgressForCurrentlyOutstandingWork,
+};
+
 @class RLMSyncUser, RLMSyncConfiguration;
 
+/**
+ The type of a progress notification block intended for reporting a session's network activity to the user.
+
+ `transferred_bytes` refers to the number of bytes that have been uploaded or downloaded.
+
+ `transferrable_bytes` refers to the number of total bytes to be uploaded or downloaded.
```

Comment:
The variable names in this documentation comment need updated to match their names in the declaration below.